src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -143,10 +143,10 @@ public AnalyzerManager(DiagnosticAnalyzer analyzer)
         /// the actions registered during <see cref="CompilationStartAnalyzerAction"/> for the given compilation.
         /// </summary>
         [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/23582", OftenCompletesSynchronously = true)]
-        public async ValueTask<AnalyzerActions> GetAnalyzerActionsAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
+        public async ValueTask<AnalyzerActions?> GetAnalyzerActionsAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
         {
             var sessionScope = await GetSessionAnalysisScopeAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
-            if (sessionScope.GetAnalyzerActions(analyzer).CompilationStartActions.Length > 0 && analyzerExecutor.Compilation != null)
+            if (sessionScope.GetAnalyzerActions(analyzer).GetValueOrDefault(AnalyzerActions.Empty).CompilationStartActionsCount > 0 && analyzerExecutor.Compilation != null)
```

Comment:
>GetValueOrDefault(AnalyzerActions.Empty) [](start = 58, length = 40)

We're doing a lot of this. Worth an extension `GetValueOrEmpty()` imo.

Reply:
This is now simplified.